client/src/image-uploader.tsx

```diff
@@ -7,25 +7,31 @@ export const ImageUploader = () => {
     const { updateAppState } = useContext(AppContext);
     const theme = useTheme();
 
-    const onDrop = useCallback((acceptedFiles: File[]) => {
-        if(acceptedFiles.length === 0)
-            return;
-
+    const onDrop = useCallback(async (acceptedFiles: File[]) => {
+        if (acceptedFiles.length === 0) return;
+    
         const file = acceptedFiles[0];
-        const blobUrl = URL.createObjectURL(file);
-
+        const preview = URL.createObjectURL(file);
+    
+        updateAppState({ 
+            image: { file, preview } 
+        });
```

Comment:
remove this. There's no image property in the AppState. Typescript must show that the type is wrong, isn't it working for you 🤔?